services/slam/client_test.go

```diff
@@ -303,6 +327,10 @@ func TestFailingClient(t *testing.T) {
 		return pFail, errors.New("failure to get position")
 	}
 
+	failingSLAMService.GetPositionFunc = func(ctx context.Context, name string) (spatial.Pose, string, error) {
+		return poseFail, componentRefFail, errors.New("failure to get position")
```

Comment:
[minor] when GetPosition fails in builtin.go, nil is returned instead of poseFail. I think we should replicate that behavior (as well as an empty string for the reference component)

Reply:
client tests are setup like this to ensure we do not pass back data when a failure occurs

Reply:
after talking offline switching these to match the newer server functions